PythonVisualizations/AdvancedSorting.py

```diff
@@ -0,0 +1,570 @@
+import random
+import time
+from tkinter import *
+try:
+    from drawable import *
+    from VisualizationApp import *
+    from SortingBase import *    
+except ModuleNotFoundError:
+    from .drawable import *
+    from .VisualizationApp import *
+    from .SortingBase import *    
+      
+
+class AdvancedArraySort(SortingBase):
+
+    def __init__(self, size=10, title="Advanced Sorting", **kwargs):
+        super().__init__(title=title, **kwargs)
+        
+        self.buttons = self.makeButtons() 
+    
+    # ARRAY FUNCTIONALITY     
+    def delete(self, val):
+        self.startAnimations()
+        callEnviron = self.createCallEnvironment()
+
+        # draw an index for variable j pointing to the first cell
+        indexDisplay = self.createIndex(0, 'j')
+        callEnviron |= set(indexDisplay)
+        
+        # go through each Drawable in the list
+        # look for val to be deleted
+        for i in range(len(self.list)):
+            n = self.list[i]
+
+            if n.val == val:
+                # get the position of the displayed cell
+                posShape = self.canvas.coords(n.display_shape)
+
+                # Highlight the found element with a circle
+                foundCircle = self.canvas.create_oval(
+                    *add_vector(
+                        posShape,
+                        multiply_vector((1, 1, -1, -1), self.CELL_BORDER)),
+                    outline=self.FOUND_COLOR)
+                callEnviron.add(foundCircle)

```

Comment:
This looks a little odd with the found oval being proportional to the value in the cell and not centered within the full cell.  Let's change `posShape` into `cellShape`, get the coordinates by calling `cellCoords()`, and pass them to the `add_vector` on line 41-42.

Reply:
Fixed in commit 1ad5d5b